sway-lib-std/src/flags.sw

```diff
@@ -1,32 +1,293 @@
+///
+/// ### Examples
+///
+/// ```sway
+/// use std::{assert::assert, flags::{disable_panic_on_unsafe_math, enable_panic_on_unsafe_math}, registers::error};
+///
+/// fn main() {
+///     disable_panic_on_unsafe_math();
+///      
+///     // Division by zero is considered unsafe math.
+///     let bar = 1 / 0; 
+///     // Error flag is set to true whenever unsafe math occurs. Here represented as 1.
+///     assert(error() == 1); 
+///
+///     enable_panic_on_unsafe_math();
+/// }
+/// ```
+pub fn disable_panic_on_unsafe_math() {
+    // Get the current value of the flags register and mask it, setting the
+    // masked bit. Flags are inverted, so set = off.
+    let flag_val = flags() | F_UNSAFEMATH_DISABLE_MASK;
+    asm(flag_val: flag_val) {
+        flag flag_val;
+    }
+}
 
+/// Enables the default `panic-on-unsafe-math` behavior in the FuelVM.
+///
+/// > **_Note:_**
+/// >
+/// > `panic-on-unsafe-math` is the default, so there is no need to use this function unless you have previously called `disable_panic_on_unsafe_math`.
+///
+/// ### Examples
+///
+/// ```sway
+/// use std::{assert::assert, flags::{disable_panic_on_unsafe_math, enable_panic_on_unsafe_math}, registers::error};
+///
+/// fn main() {
+///     disable_panic_on_unsafe_math();
+///      
+///     // Division by zero is considered unsafe math.
+///     let bar = 1 / 0; 
+///     // Error flag is set to true whenever unsafe math occurs. Here represented as 1.
+///     assert(error() == 1); 
+///
+///     enable_panic_on_unsafe_math();
+/// }
+/// ```
+pub fn enable_panic_on_unsafe_math() {
     // Get the current value of the flags register and mask it, unsetting the
     // masked bit. Flags are inverted, so unset = on.
-    let flag_val = flags() & mask;
+    let flag_val = flags() & F_UNSAFEMATH_ENABLE_MASK;
     asm(flag_val: flag_val) {
         flag flag_val;
     }
 }
+
+/// Allows unsafe math operations to occur without a FuelVM panic.
+/// More suitable than `disable_panic_on_unsafe_math` for use in functions that are not the entry point of a program.
+///
+/// > **_WARNING:_**
+/// >
+/// > Don't forget to call `set_flags` after performing the operations for which you disabled the default `panic-on-unsafe-math` behavior in the first place!
+///
+/// ### Returns
+/// 
+/// * [u64] - The flag prior to disabling panic on overflow.
+///
+/// ### Examples
+///
+/// ```sway
+/// use std::{assert::assert, flags::{disable_panic_on_unsafe_math_preserving, set_flags}, registers::error};
+///
+/// fn foo() {
+///     let prior_flags = disable_panic_on_unsafe_math_preserving();
+///      
+///     // Division by zero is considered unsafe math.
+///     let bar = 1 / 0; 
+///     // Error flag is set to true whenever unsafe math occurs. Here represented as 1.
+///     assert(error() == 1); 
+///
+///     set_flags(prior_flags);
+/// }
+/// ```
+pub fn disable_panic_on_unsafe_math_preserving() -> u64 {
```

Comment:
Why the "*_preserving" variants are needed? Can´t we just always do this and ignore the result when not needed?